[CRUD] PART 1: Setup POST Product Detail Page

Taliko5 opened the pull request:
this PR does implementing below  
-introduce  post route method  (productDetail )
-import additional material UI 
-introducing creating new product (AddProduct)

Fixes Github issue #11 

### Please do not merge. Merci. :)

Commit:
post

Commit:
create availability(date)

Commit:
change warrantyPeriod to date"

Commit:
fixed avaiability

Commit:
fixed currency

Commit:
fixed currency

Commit:
trying imageupdate

Commit:
button submit  addproduct

Commit:
finish post

1000miles committed:
Merge branch 'master' of github.com:IchiBizz/IchiBizz into noriko-create-productdetail-post-method

1000miles committed:
Rename Product props based on recent Product change

- adopt changes in components/AddProduct.js
- adopt changes in routes/productDetail.js
- TOFIX: After authentication setup ix seller/owner prop

1000miles committed:
More props renaming based on product model

1000miles commented:
As discussed with @noliko-zwa I am taking over this PR now.

### Please do not merge. Merci. :)

1000miles committed:
Clean up: Add Product

- remove wishlist and requested props that is not possible
  on product CREATE method
- comment out parts to be modified (price & currency, maps)

1000miles commented:
@ninetteadhikari 

I realized that it is a bit messy/confusing to continue on Noriko's branch `noriko-create-productdetail-post-method` so I will check out into a separate branch from Noriko's branch.

So far creating a product via postman works with all properties except the dates.

```json
{
    "imageUrl": [
        "https://picsum.photos/id/1/200/300"
    ],
    "tags": [
        "office, food"
    ],
    "wishlist": [],
    "requested": [],
    "_id": "5d98612650f5ad18d059b146",
    "title": "Renault 10",
    "description": "This is great work all!",
    "brand": "Siemens",
    "category": "restaurant",
    "quantity": 5,
    "price": 345,
    "company": "lalalala",
    "isSold": false,
    "__v": 0
}
```

@ninetteadhikari Do you mind to review this PR and merge it soon if your tests are successful? Thank you. 

cc @noliko-zwa

1000miles commented:
Strange, it works now, it just took much longer via postman. Also posting a date for `availability` and `warrantyUntil` works now.

```json
{
    "imageUrl": [
        "https://picsum.photos/id/1/200/300"
    ],
    "tags": [
        "office, food"
    ],
    "wishlist": [],
    "requested": [],
    "_id": "5d9863c150f5ad18d059b149",
    "title": "Renault 10",
    "description": "This is great work all!",
    "brand": "Siemens",
    "category": "restaurant",
    "quantity": 5,
    "price": 345,
    "company": "lalalala",
    "availability": "2019-10-05T09:34:02.834Z",
    "warrantyUntil": "2019-10-05T09:34:02.834Z",
    "isSold": false,
    "__v": 0
}
```

ninetteadhikari commented on client/src/components/AddProduct.js:
```diff
@@ -0,0 +1,349 @@
+      longitude: null
+    },
+    // Must be future date
+    availability: new Date(),
```
what do you think about having default date as 'null' for availability and warrantyUntil ? This way if user doesn't choose anything no date is saved instead of 'today's date'

ninetteadhikari commented on client/src/components/AddProduct.js:
```diff
@@ -0,0 +1,349 @@
+        currency: this.state.currency,
+        company: this.state.company,
+        location: {
+          latitude: this.state.latitude,
```
not sure if lat long is just this.state.latitude or this.state.location.latitude. Just a comment that this needs to be verified once the GoogleMap input it setup

ninetteadhikari commented on routes/productDetail.js:
```diff
@@ -0,0 +1,59 @@
+    currency,
+    tags,
+    company,
+    latitude,
```
this needs to be modified according to current model schema where lat long is nested under location. I can look at this once google maps is setup

ninetteadhikari commented on routes/productDetail.js:
```diff
@@ -0,0 +1,59 @@
+    currency: currency,
+    tags: tags,
+    company: company,
+    latitude: latitude,
```
this needs to be modified according to current model schema where lat long is nested under location. I can look at this once google maps is setup

1000miles reviewed (approved):
Thanks for the comments, @ninetteadhikari. Good catch, let's keep the minor checks and changes for later and merge now.

ninetteadhikari commented:
I reviewed and the comments are more of reminders to be fixed with later features:
- Update lat, long for both state and pot route file to match the model

This is not a blocker for merging to master.
